SConstruct

```diff
@@ -1,93 +1,134 @@
+        if env['target'] in ('debug', 'd'):
+            env.Append(CCFLAGS = ['-EHsc', '-D_DEBUG', '-MDd'])
+        else:
+            env.Append(CCFLAGS = ['-O2', '-EHsc', '-DNDEBUG', '-MD'])
+    # untested
+    else:
+        if env['target'] in ('debug', 'd'):
+            env.Append(CCFLAGS = ['-fPIC', '-g3','-Og', '-std=c++17'])
+        else:
+            env.Append(CCFLAGS = ['-fPIC', '-g','-O3', '-std=c++17'])
+
+# untested
+elif env['platform'] == 'osx':
+    env['target_path'] += 'osx/'
+    godot_cpp_library += '.osx'
     platform_dir = 'osx'
-    godot_openvr_path = godot_openvr_path + 'osx/'
-    env.Append(CCFLAGS = ['-g','-O3', '-arch', 'x86_64'])
-    env.Append(CXXFLAGS='-std=c++11')
-    env.Append(LINKFLAGS = ['-arch', 'x86_64'])
+    if env['target'] in ('debug', 'd'):
+        env.Append(CCFLAGS = ['-g','-O2', '-arch', 'x86_64'])
+        env.Append(LINKFLAGS = ['-arch', 'x86_64'])
+    else:
+        env.Append(CCFLAGS = ['-g','-O3', '-arch', 'x86_64'])
+        env.Append(LINKFLAGS = ['-arch', 'x86_64'])
 
-elif platform == "linux":
+elif env['platform'] in ('x11', 'linux'):
+    env['target_path'] += 'x11/'
+    godot_cpp_library += '.linux'
     platform_dir = 'linux'
-    godot_openvr_path = godot_openvr_path + 'x11/'
-    env.Append(CCFLAGS = ['-fPIC', '-g','-O3', '-std=c++14'])
-    env.Append(CXXFLAGS='-std=c++0x')
-    env.Append(LINKFLAGS = ['-Wl,-R,\'$$ORIGIN\''])
-
-elif platform == "windows":
-    platform_dir = 'win'
-    godot_openvr_path = godot_openvr_path + 'win' + str(bits) + '/'
-    if target == "debug":
-        env.Append(CCFLAGS = ['-EHsc', '-D_DEBUG', '/MDd'])
+    if env['target'] in ('debug', 'd'):
+        env.Append(CCFLAGS = ['-fPIC', '-g3','-Og', '-std=c++17'])
     else:
-        env.Append(CCFLAGS = ['-O2', '-EHsc', '-DNDEBUG', '/MD'])
+        env.Append(CCFLAGS = ['-fPIC', '-g','-O3', '-std=c++17'])
+
+# Complete godot-cpp library path
+if env['target'] in ('debug', 'd'):
+    godot_cpp_library += '.debug'
+else:
+    godot_cpp_library += '.release'
+
+godot_cpp_library += '.' + str(env['bits'])
 
-# add our openvr library
-platform_dir += str(bits)
+# Update our include search path 
+env.Append(CPPPATH=[
+    '.', 
+    'src/', 
+    'src/open_vr/', 
+    godot_headers_path,
+    godot_cpp_path + 'include/',
+    godot_cpp_path + 'include/core/',
+    godot_cpp_path + 'include/gen/',
+    env['openvr_path'] + 'headers/'])
 
-env.Append(CPPPATH=[openvr_path + 'headers/'])
+# Add our godot-cpp library
+env.Append(LIBPATH=[godot_cpp_path + 'bin/'])
+env.Append(LIBS=[godot_cpp_library])
+
+# Add our openvr library
+platform_dir += str(env['bits'])
 
 if (os.name == "nt" and os.getenv("VCINSTALLDIR")):
-    env.Append(LIBPATH=[openvr_path + 'lib/' + platform_dir])
+    env.Append(LIBPATH=[env['openvr_path'] + 'lib/' + platform_dir])
     env.Append(LINKFLAGS=['openvr_api.lib'])
 elif platform == "osx":
```

Comment:
Got an Scons error here. I think this line should be

`elif env['platform'] == "osx":`

Reply:
I think so too :)